scripts/commands/checkmission.lua

```diff
@@ -13,7 +13,20 @@ cmdprops =
 
 function onTrigger(player,logId,target)
     
+    local logName;
     logId = tonumber(logId) or _G[logId];
+    if ((type(logId) == "table")) then
+        logName = logId.full_name;
+        logId = logId.mission_log;
+    else
+        if (logId <= 2) then
+            local logNames = {"San d'Oria", "Bastok", "Windurst"}
```

Comment:
I think those strings were slapped in because prior to #3322 the nation variables were still integers due to conquest scripts. So a command of `@checkmission player SANDORIA` would always come in as an integer, and didn't have a full name in a would-be table to grab. Hence quick if check with local array of strings just to get the "San d'Oria" for the "Current San d'Oria Mission is 5".

Although since they're actually tables now, `SANDORIA`, `BASTOK`, and `WINDURST` could probably be tossed into the missionAreas array and used like all the others to fetch a full log name, killing the `if (logId <= 2)` check entirely.

Regarding killing case sensitivity, I think each QMF area is its own variable and isn't part of a parent, so key-based reference (ie: `log_information[string.upper(text_from_gm)]`) might need a lookup table added
```
log_information = {
    ["SANDORIA"] = SANDORIA
```
to link a given string to a table.

Another potential issue is that I think at the moment all the GM commands just treat/invoke the argument as if they were a raw variable (ie: literally passing the `ADOULIN` table instead of a string of "ADOULIN").